File: day14/part2.go
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"strings"
)

func main() {
	file, _ := ioutil.ReadFile("input")
	input := []byte(strings.TrimSpace(string(file)))

	recipes := []byte("37")
	p1 := 0
	p2 := 1

	done := false
	for !done {
		n1 := recipes[p1] - '0'
		n2 := recipes[p2] - '0'
		sum := n1 + n2
		var num []byte
		if sum < 10 {
			num = []byte{sum}
		} else {
			num = []byte{1, sum - 10}
		}
		for _, n := range num {
			recipes = append(recipes, n + '0')
			if len(recipes) > len(input) && bytes.Equal(recipes[len(recipes)-1-len(input):len(recipes)-1], input) {
				done = true
				break
			}
		}
		p1 = (p1 + 1 + int(n1)) % len(recipes)
		p2 = (p2 + 1 + int(n2)) % len(recipes)
	}
	fmt.Println(len(recipes) - len(input) - 1)
}

File: day14/part2.js
var fs = require('fs')
 
let input = String(fs.readFileSync('input')).trim().split('')
for (var i = 0; i < input.length; i++) {
    input[i] = input[i].charCodeAt(0)-'0'.charCodeAt(0)
}

let recipes = [3, 7]
let p1 = 0
let p2 = 1

let done = false
while (!done) {
  let n1 = recipes[p1]
  let n2 = recipes[p2]
  let sum = n1 + n2
  if (sum<10) {
    num = [sum]
  } else {
    num = [1, sum-10]
  }
  for (let i = 0; i < num.length; i++) {
    recipes.push(num[i])
    for (var j = 0; j<input.length; j++) {
        if (recipes[recipes.length-1-j]!=input[input.length-1-j]) {
            break
        }
    }
    if (j==input.length) {
        done = true
        break
    }
  }
  p1 = (p1 + 1 + n1) % recipes.length
  p2 = (p2 + 1 + n2) % recipes.length
}
console.log(recipes.length - input.length)